docs/skaffold.md

```diff
@@ -0,0 +1,39 @@
+## Development with skaffold
+
+The network operator includes a `skaffold.yaml` for iterative development with `skaffold`.
+
+For more information on `skaffold` [see the official docs](https://skaffold.dev/).
+
+### Prerequisites
+NOTE: This list does not include dependencies of the `network-operator` e.g. Node Feature Discovery or cert-manager. These must be installed on the cluster independently where required.
+
+- Skaffold CLI - https://skaffold.dev/docs/install/#standalone-binary
+- An image registry to push images to.
+- kustomize
+- kubectl
+- Docker
+- A running Kubernetes cluster.
+
+### Deploying with Skaffold
+
+Export a registry to push images to:
+
+`export REGISTRY="localhost:5000`
+
+From the root directory of the `network-operator` repository run:
+
+`skaffold dev --default-repo $REPOSITORY`
```

Comment:
you may want --trigger=manual so that it does not rebuild on each change, WDTY?

perhaps add some example with minikube

```
minikube --profile netop-dev start
# this one will point docker cli to the same daemon as used by minikube nfd-dev cluster
# that way skaffold will build images in the correct local repo.
eval $(minikube -p nfd-dev docker-env)
# deploy network attachment definition crds
kubectl create -f deployment/network-operator/crds/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml
# label nodes (may need more labels to get mofed container to deploy on it like os / arc/ kernel etc)
kubectl label node netop-dev node-role.kubernetes.io/worker=
kubectl label node netop-dev feature.node.kubernetes.io/pci-15b3.present=true
skaffold dev --trigger=manual

# Cleanup
minikube --profile netop-dev delete
```

i tried setting up minikube with 2 nodes (`minikube --profile netop-dev start -n 2`) but seems that need to setup minikube registry addon for it so for now we can start with one node :)

Reply:
we should then have a dev env where one can create a nicclusterpolicy object and actual operands will be deployed e.g mofed container

Reply:
Sounds good! I'll push the minikube stuff to a second iteration though if you don't mind - the skaffold stuff is really useful to me running on full clusters as-is.

Reply:
> I think for dev this is probably desired behaviour - at least from my POV. If you feel strongly I can definitely add the flag though.

up 2 u, me personally im heavy fingered on the ctrl+s :) so i dont really want to redeploy each time.

Reply:
> Planning on tackling minikube / some other lightweight dev setup in another iteration.

Ack, we got #576 for tracking this.

Reply:
btw without:
`kubectl create -f deployment/network-operator/crds/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml` it will not work so perhaps mention this in doc.

Reply:
> btw without:
kubectl create -f deployment/network-operator/crds/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml it will not work so perhaps mention this in doc.

Thanks for catching that - I've added that step to the `skaffold.yaml`